subsys/bluetooth/controller/ll_sw/ll_adv.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2017 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+struct ll_adv_set {
+	u8_t  chl_map:3;
+	u8_t  filter_policy:2;
+#if defined(CONFIG_BLUETOOTH_CONTROLLER_PRIVACY)
+	u8_t      rl_idx : 5;
+	u8_t      own_addr_type : 2;
+	u8_t      id_addr_type : 1;
+	u8_t      id_addr[BDADDR_SIZE];
```

Comment:
Having both little and big endian variants to ensure the the same memory layout makes only sense if we care about the exact memory layout. Unless I misunderstood something, in this case we don't care about the layout since these variables are only for local storage like any other non-packed struct and the less-than-8-bits limitation is there mainly for convenience so we don't have to add explicit masking when using the variables.